haystack/pipeline.py

```diff
@@ -136,6 +141,100 @@ def draw(self, path: Path = Path("pipeline.png")):
         graphviz.layout("dot")
         graphviz.draw(path)
 
+    @classmethod
+    def load_from_yaml(cls, path: Path, pipeline_name: Optional[str] = None):
+        """
+        Load Pipeline from a YAML file defining the individual components and how they're tied together to form
+        a Pipeline. A single YAML can declare multiple Pipelines, in which case an explicit `pipeline_name` must
+        be passed.
+
+        :param path: path of the YAML file.
+        :param pipeline_name: if the YAML contains multiple pipelines, the pipeline_name to load must be set.
+        """
+        with open(path, "r") as stream:
+            data = yaml.safe_load(stream)
+
+        if pipeline_name is None:
+            if len(data["pipelines"]) == 1:
+                pipeline_config = data["pipelines"][0]
+            else:
+                raise Exception("The YAML contains multiple pipelines. Please specify the pipeline name to load.")
+        else:
+            pipelines_in_yaml = list(filter(lambda p: p["name"] == pipeline_name, data["pipelines"]))
+            if not pipelines_in_yaml:
+                raise Exception(f"Cannot find any pipeline with name '{pipeline_name}' declared in the YAML file.")
+
+        definitions = {}  # definitions of each component from the YAML.
+        for definition in data["components"]:
+            cls._interpolate_environment_variables(definition)
+            name = definition.pop("name")
+            definitions[name] = definition
+
+        pipeline = cls()
+
+        components: dict = {}  # instances of component objects.
+        for node_config in pipeline_config["nodes"]:
+            name = node_config["name"]
+            component = cls._load_or_get_component(name=name, definitions=definitions, components=components)
+            if "DocumentStore" not in definitions[name]["type"]:  # DocumentStore is not an explicit node in a Pipeline
+                pipeline.add_node(component=component, name=node_config["name"], inputs=node_config["inputs"])
```

Comment:
Does the init order impact us somehow here? For example, if the YAML defines component A before component B , but A has B as `inputs` :thinking:

Reply:
There are two parts here: the ordering of `components` & the ordering of `nodes` in a pipeline.

The order in which `components` are defined does _not_ matter, while, the `nodes` is a strictly ordered list.  

While loading a pipeline from YAML, we iterative over the `nodes` to load the corresponding components. Since each node in the `inputs` must be defined previously in the pipeline, we can be sure it has already been loaded.

Another benefit of iterating over `nodes` is we only load the components that are needed. For instance, a YAML may contain components for 2 pipelines, however, when loading a single pipeline not all components needs to be loaded.

Reply:
yeah, I got the difference between handling `components` and `nodes`. Makes totally sense. Just wondered about the order of nodes  in the yaml - but if I got you right, the order of `nodes` in the YAML actually matters.

Reply:
Yes, the order of `nodes` matter. The behaviour is similar to creating a Pipeline instance in Python code.